.core_files.yaml

```diff
@@ -79,6 +79,7 @@ components: &components
   - homeassistant/components/group/**
   - homeassistant/components/hassio/**
   - homeassistant/components/homeassistant/**
+  - homeassistant/components/homeassistant_hardware/**
```

Comment:
I added this to `.core_files.yaml` as multiple other components also depend on it.

Reply:
Moved to #130129

Reply:
`ci-full-run` label doesn't take effect on "re-run all jobs", so I will update after #130129 is merged.